src/rpcmining.cpp

```diff
@@ -45,9 +45,13 @@ Value GetNetworkHashPS(int lookup, int height) {
     if (pb == NULL || !pb->nHeight)
         return 0;
 
-    // If lookup is -1, then use blocks since last difficulty change.
+    // If lookup is -1, then use difficulty averaging window.
     if (lookup <= 0)
-        lookup = pb->nHeight % Params().GetConsensus().DifficultyAdjustmentInterval() + 1;
+        lookup = pb->nHeight - Params().GetConsensus().nPowAveragingWindow;
+
+    // If lookup is still negative, then use blocks since genesis.
+    if (lookup <= 0)
+        lookup = pb->nHeight;
```

Comment:
Shouldn't this be ``if (lookup < 0) lookup = 0;``? I.e. of there are not enough blocks for the full window, we should still use however many blocks are available.